startup.py

```diff
@@ -0,0 +1,331 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import _winreg
+
+from xml.etree import ElementTree
+
+from sgtk import TankError
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+
+class MaxLauncher(SoftwareLauncher):
+    """
+    Handles launching 3dsMax executables. Automatically starts up
+    a tk-3dsmaxplus engine with the current context in the new session
+    of 3dsMax.
+    """
+
+    @property
+    def minimum_supported_version(self):
+        """
+        The minimum software version that is supported by the launcher.
+        """
+        return "2015"
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param list versions: List of strings representing versions
+                              to search for. If set to None, search
+                              for all versions. A version string is
+                              DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52".
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        # First look for executables using the Autodesk Synergy registry.
+        sw_versions = self._synergy_software_versions(versions)
+        if not sw_versions:
+            sw_versions = self._default_path_software_versions(versions)
+
+        if not sw_versions:
+            self.logger.info("Unable to determine available SoftwareVersions for engine %s" % self.engine_name)
+            return []
+
+        return sw_versions
+
+    def prepare_launch(self, exec_path, args, file_to_open=None):
+        """
+        Prepares an environment to launch 3dsMax in that will automatically
+        load Toolkit and the tk-3dsmaxplus engine when 3dsMax starts.
+
+        :param str exec_path: Path to 3dsMax executable to launch.
+        :param str args: Command line arguments as strings.
+        :param str file_to_open: (optional) Full path name of a file to open on launch.
+        :returns: :class:`LaunchInformation` instance
+        """
+        required_env = {}
+
+        startup_script_path = os.path.join(self.disk_location, "plugins", "basic", "bootstrap", "(zero-config)", "bootstrap.ms")
```

Comment:
i guess we'll see later on if it makes more sense to keep a single startup file (e.g. in `tk-3dsmaxplus/startup`) or separate ones for classic and plugins!